python-package/lightgbm/basic.py

```diff
@@ -2144,16 +2186,29 @@ def __init__(self, params=None, train_set=None, model_file=None, model_str=None,
                 if alias in params:
                     machines = params[alias]
                     if isinstance(machines, str):
-                        num_machines = len(machines.split(','))
+                        num_machines_from_machine_list = len(machines.split(','))
                     elif isinstance(machines, (list, set)):
-                        num_machines = len(machines)
+                        num_machines_from_machine_list = len(machines)
                         machines = ','.join(machines)
                     else:
                         raise ValueError("Invalid machines in params.")
-                    self.set_network(machines,
-                                     local_listen_port=params.get("local_listen_port", 12400),
-                                     listen_time_out=params.get("listen_time_out", 120),
-                                     num_machines=params.setdefault("num_machines", num_machines))
+
+                    params = _choose_param_value(
+                        main_param_name="num_machines",
+                        params=params,
+                        default_value=num_machines_from_machine_list
+                    )
+                    params = _choose_param_value(
+                        main_param_name="local_listen_port",
+                        params=params,
+                        default_value=12400
+                    )
+                    self.set_network(
+                        machines=machines,
+                        local_listen_port=params["local_listen_port"],
+                        listen_time_out=params.get("listen_time_out", 120),
+                        num_machines=params["num_machines"]
+                    )
                     break
```

Comment:
I think we should remove all aliases of `machines` from `params`. This `break` prevent doing this.

Reply:
ok, I proposed something for this in https://github.com/microsoft/LightGBM/pull/3813/commits/63387db2992ce13f1490a8b1202e30e21142190f

I think the cleanest way to do it is to use `_choose_param_value()` to get a single value for `machines`. Then we don't need a separate `for` loop with a `break`.